apps/src/feedback.js

```diff
@@ -790,6 +781,28 @@ FeedbackUtils.prototype.getEmptyContainerBlock_ = function() {
 };
 
 /**
+ * Check for empty container blocks, and return an appropriate failure
+ * code if any are found.
+ * @return {TestResults} ALL_PASS if no empty blocks are present, or
+ *   EMPTY_BLOCK_FAIL or EMPTY_FUNCTION_BLOCK_FAIL if empty blocks
+ *   are found.
+ */
+FeedbackUtils.prototype.checkForEmptyContainerBlockFailure_ = function() {
+  var emptyBlock = this.getEmptyContainerBlock_();
+  if (emptyBlock) {
+    var type = emptyBlock.type;
+    if (type === 'procedures_defnoreturn' || type === 'procedures_defreturn') {
+      return TestResults.EMPTY_FUNCTION_BLOCK_FAIL;
+    }
+    // Block is assumed to be "if" or "repeat" if we reach here.
+    // This is where to add checks if you want a different TestResult
+    // for "controls_for_counter" blocks, for example.
+    return TestResults.EMPTY_BLOCK_FAIL;
+  }
+  return TestResults.ALL_PASS;
+};
```

Comment:
Nit: could invert this: if (!emptyBlock) { return ALL_PASS; } and remove a level of indentation.